pandas/tests/indexes/test_base.py

```diff
@@ -871,6 +871,18 @@ def test_sub(self):
         pytest.raises(TypeError, lambda: index - index.tolist())
         pytest.raises(TypeError, lambda: index.tolist() - index)
 
+    def test_sub_object(self):
+        # GH#19369
+        from decimal import Decimal
+        index = pd.Index([Decimal(1), Decimal(2)])
+        expected = pd.Index([Decimal(0), Decimal(1)])
+
+        result = index - Decimal(1)
+        tm.assert_index_equal(result, expected)
+
+        result = index - pd.Index([Decimal(1), Decimal(1)])
+        tm.assert_index_equal(result, expected)
+
```

Comment:
Can we also add tests where element-wise subtraction should fail.

Reply:
There are already some in the test just above this, but more couldn’t hurt.  Will do.

Reply:
Ah, yes, thanks for pointing that out.  Two things would be good there:

* We should rename that test above to be `test_sub_fail`
* Add a couple of tests where element-wise subtraction should fail

Reply:
You got it.